.github/workflows/ci-cd.yml

```diff
@@ -311,3 +311,118 @@ jobs:
         run: |
           gh release upload --repo vacanza/holidays ${{ github.ref_name }} dist/*
           gh release upload --repo vacanza/holidays ${{ github.ref_name }} holidays-*-sbom.json
+
+  upload-to-zenodo:
+    name: Publish release to Zenodo
+    if: github.event_name == 'release' && github.event.action == 'published'
+    runs-on: ubuntu-latest
+    needs: [update-github-release]
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download SBOM
+        uses: actions/download-artifact@v4
+        with:
+          name: sbom
+
+      - name: Download signed artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: signed-artifacts
+          path: dist
+
+      - name: Prepare release archive
+        run: |
+          zip -r vacanza-holidays-${{ github.ref_name }}.zip dist/* holidays-*-sbom.json
+
+      - name: Check for existing Zenodo deposition
```

Comment:
It looks like we should create our own action for this, it seems https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action would fit.

If possible let's have it in Python